packages/webapp/src/_app/utils/asset.ts

```diff
@@ -5,6 +5,14 @@ export interface Asset {
 }
 
 export const assetFromString = (asset: string): Asset => {
+    const re = /^[0-9]+\.[0-9]+ [A-Z,a-z]{1,5}$/
+    if(asset.match(re) === null) {
+        return {
+            symbol: "ERR",
+            quantity: 0.0,
+            precision: 2
+        }
+    }
```

Comment:
Verify the format to ensure we successfully parse it, and provide obvious feedback to the user in case of error

Reply:
I like that you're adding validation here.

As for what to do when the validation fails, I think I'd be more comfortable `throw`ing an error here rather than returning something to the UI/view layer that acts like everything's okay. It should be the job of the UI layer to decide what to display to the user in the event of an exception.

Reply:
...that said, I'm okay with the idea of doing that in a follow-up PR, since we'll need to handle that new exception in various other places in the code...and that might blow the scope of this one.

Reply:
@brandonfancher I'd propose just doing it now, given this is a case we're already not handling in any way. Might as well do this part right and run into what we're going to run into anyway and have this part right. Lmk if you disagree; otherwise, that'll be my plan.

Reply:
Adding the throw in there is fine. But we'll also want to wrap any invocation of this function in a `try...catch` too. Otherwise, we may run into unhandled exceptions.